integ-test/src/test/java/org/opensearch/sql/sql/DateTimeFunctionIT.java

```diff
@@ -225,6 +225,17 @@ public void testDayOfYear() throws IOException {
     verifyDataRows(result, rows(260));
   }
 
+  @Test
+  public void testDay_Of_Year() throws IOException {
+    JSONObject result = executeQuery("select day_of_year(date('2020-09-16'))");
+    verifySchema(result, schema("day_of_year(date('2020-09-16'))", null, "integer"));
+    verifyDataRows(result, rows(260));
+
+    result = executeQuery("select day_of_year('2020-09-16')");
+    verifySchema(result, schema("day_of_year('2020-09-16')", null, "integer"));
+    verifyDataRows(result, rows(260));
+  }
```

Comment:
Can we please add a datetime object passed to day_of_year.

Reply:
Also add a time object passed to day_of_year (does it convert to datetime first?)

Reply:
Try with all datetime types. `TIME` depends on https://github.com/opensearch-project/sql/pull/1047 though.

Reply:
Added in https://github.com/Bit-Quill/opensearch-project-sql/pull/173/commits/4b09fb86c531a619961d6d6691268e97948e67d9